modules/nixos/general/keyboard.nix

```diff
@@ -0,0 +1,54 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.mine.keyboard;
+
+  layouts = {
+    azerty = {
+      layout = "fr";
+      variant = "";
+    };
+    bepo = {
+      layout = "fr";
+      variant = "bepo";
+    };
+    colemak = {
+      layout = "us";
+      variant = "colemak";
+    };
+    qwerty = {
+      layout = "us";
+      variant = "";
+    };
+    qwerty_intl = {
+      layout = "us";
+      variant = "intl";
+    };
+  };
+in
+
+with lib;
+
+{
+  options = {
+    mine.keyboard = {
+      layouts = mkOption {
+        type = types.listOf (types.enum (attrNames layouts));
+        default = [ "qwerty" ];
+        example = [ "colemak" "qwerty" ];
+        description = "Layouts to set";
+      };
+
+      enableAtBoot = mkEnableOption "If true, the first of the layouts will be enabled at boot";
+    };
+  };
+
+  config = {
+    boot.earlyVconsoleSetup = cfg.enableAtBoot;
+    i18n.consoleKeyMap = if layouts.(builtins.head cfg.layouts).variant == ""
+                         then layouts.(builtins.head cfg.layouts).layout
+                         else layouts.(builtins.head cfg.layouts).variant;
+    services.xserver.layout = toString (map (n: layouts.n.layout) cfg.layouts);
```

Comment:
This should be a comma-separated list of layouts.

https://github.com/NixOS/nixpkgs/blob/3366e982e8143e2b8fa5e02818ed86d4ec3ba918/nixos/modules/services/x11/xserver.nix#L311-L317